### PR TITLE
fix(condo): DOMA-6059 Remove RecurrentPaymentContext in case if user, resident, serviceConsumer was deleted

### PR DESCRIPTION
**Pull request by ekabardinsky**

### CODE REVIEW

File: apps/condo/domains/resident/schema/Resident.js [hunk 1]
```diff
@@ -250,6 +252,25 @@ const Resident = new GQLListSchema('Resident', {
...
+                const recurrentContexts = await RecurrentPaymentContext.getAll(context, {
+                    serviceConsumer: {
+                        resident: {
+                            id: updatedItem.id,
+                        },
+                    },
+                })
+
+                for (const recurrentContext of recurrentContexts) {
+                    if (!recurrentContext.deletedAt) {
```
Comment (pahaz) [inline apps/condo/domains/resident/schema/Resident.js:269]: it's better to add deletedAt as filter condition an line 262

File: apps/condo/domains/resident/schema/ServiceConsumer.js [hunk 2]
```diff
@@ -122,6 +123,30 @@ const ServiceConsumer = new GQLListSchema('ServiceConsumer', {
...
     },
+    hooks: {
+        afterChange: async ({ context, operation, updatedItem, originalInput }) => {
+            const deletedAt = get(updatedItem, 'deletedAt', null)
+            const sender = get(originalInput, 'sender', null)
+            const dv = get(originalInput, 'dv', null)
+
+            // handle soft delete
+            // in order to soft delete recurrent payment contexts
+            if (operation === 'update' && deletedAt) {
```
Comment (pahaz) [inline apps/condo/domains/resident/schema/ServiceConsumer.js:134]: may be you need to extract some serviceSchema helter for that. Look at Ticket hook handlers examples

File: apps/condo/domains/resident/schema/Resident.js (hunk 1 above)
Comment (ekabardinsky) [reply apps/condo/domains/resident/schema/Resident.js]: I have extracted function for delete orphans recurrentContexts into acquiring/utils/serverSchema/helpers.js
The function includes this change

File: apps/condo/domains/resident/schema/ServiceConsumer.js (hunk 2 above)
Comment (ekabardinsky) [reply apps/condo/domains/resident/schema/ServiceConsumer.js:134]: I have extracted function for delete orphans recurrentContexts into acquiring/utils/serverSchema/helpers.js

File: apps/condo/domains/acquiring/utils/serverSchema/helpers.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+ * @param userId (id, optional)
+ * @param residentId (id, optional)
+ * @param serviceConsumerId (id, optional)
+ * @param dv
+ * @param sender
+ * @returns {Promise<void>}
+ */
+const removeOrphansRecurrentPaymentContexts = async ({ context, userId, residentId, serviceConsumerId, dv, sender }) => {
+    if (isNil(userId) &&  isNil(residentId) && isNil(serviceConsumerId))
+        throw new Error('Can not removeOrphansRecurrentPaymentContexts for empty parent ids')
```
Comment (pahaz) [inline apps/condo/domains/acquiring/utils/serverSchema/helpers.js:17]:
```suggestion
    if (isNil(userId) &&  isNil(residentId) && isNil(serviceConsumerId)) throw new Error('Can not removeOrphansRecurrentPaymentContexts for empty parent ids')
```

File: apps/condo/domains/acquiring/utils/serverSchema/helpers.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+    if (isNil(dv) ||  isNil(sender))
+        throw new Error('Can not removeOrphansRecurrentPaymentContexts with empty dv and sender params')
```
Comment (pahaz) [inline apps/condo/domains/acquiring/utils/serverSchema/helpers.js:20]:
```suggestion
    if (isNil(dv) ||  isNil(sender)) throw new Error('Can not removeOrphansRecurrentPaymentContexts with empty dv and sender params')
```

**Review by pahaz — APPROVED** [review]

File: apps/condo/domains/acquiring/utils/serverSchema/helpers.js (hunk 4 above)
Comment (ekabardinsky) [reply apps/condo/domains/acquiring/utils/serverSchema/helpers.js]: ok

**Pull request merged**